src/blockchains/cosmos.ts

```diff
@@ -0,0 +1,81 @@
+import { BlockchainHandler, BlockchainHandlerOpts } from '../blockchain-handler';
+import { AccountID } from 'caip';
+import { getConsentMessage, LinkProof } from '../utils';
+import { signTx, Tx, SignMeta, verifyTx } from '@tendermint/sig';
+const { base64ToBytes, bytesToBase64, toCanonicalJSONBytes } = require('@tendermint/belt');
+import * as uint8arrays from 'uint8arrays';
+
+const namespace = 'cosmos';
+
+const stringEncode = (str: string): string => bytesToBase64(uint8arrays.fromString(str));
+
+// return data in the cosmos unsigned transaction format
+function asTransaction(address: string, message: string): Tx {
+  return {
+    fee: {
+      amount: [{ amount: '0', denom: '' }],
+      gas: '0',
+    },
+    memo: message,
+    msg: [
+      {
+        type: 'cosmos-sdk/MsgSend',
+        value: {
+          from_address: address,
+          to_address: address,
+          amount: [{ amount: '0', denom: '0' }],
+        },
+      },
+    ],
+  };
+}
+
+// generate metadata for signing the transaction
+function getMetaData(): SignMeta {
+  return {
+    account_number: '1',
+    chain_id: 'cosmos',
+    sequence: '0',
+  };
+}
+
+async function createLink(did: string, account: AccountID, provider: any, opts: BlockchainHandlerOpts): Promise<LinkProof> {
+  const { message, timestamp } = getConsentMessage(did, !opts?.skipTimestamp);
+  const encodedMsg = stringEncode(message);
+  const res = await signTx(asTransaction(account.address, encodedMsg), getMetaData(), provider);
+  const signature = bytesToBase64(toCanonicalJSONBytes(res.signatures[0]));
+  const proof: LinkProof = {
+    version: 2,
+    type: 'eoa',
+    message: message,
+    signature,
+    account: account.toString(),
+  };
+  if (!opts?.skipTimestamp) proof.timestamp = timestamp;
+  return proof;
+}
+
+async function authenticate(message: string, account: AccountID, provider: any): Promise<string> {
+  const encodedMsg = stringEncode(message);
+  const res = await signTx(asTransaction(account.address, encodedMsg), getMetaData(), provider);
+  return bytesToBase64(toCanonicalJSONBytes(res.signatures[0]));
```

Comment:
could we hash (sha256 like eth implementation) the signature output here? so that a fixed length hex string is returned